### PR TITLE
Cap LDAP user cache

### PR DESCRIPTION
**Pull request by butonic**
Caps memory usage when iterating over all users, eg in a sync job

Before

6-ldap

After

7-ldapcapped

The memdump is from 9.0 without some of the memory pathes. Nevertheless, it should be recognizable that the createAndCache method has shrunk considerably.

### CODE REVIEW

**Comment by PVince81** [comment]
@owncloud/ldap

File: apps/user_ldap/lib/User/Manager.php [hunk 1]
```diff
@@ -62,14 +63,13 @@ class Manager {
...
-	 * array['byDN']	\OCA\User_LDAP\User\User[]
-	 * 	['byUid']	\OCA\User_LDAP\User\User[]
-	 * @var array $users
+	 * @var CappedMemoryCache $usersByDN
 	 */
-	protected $users = array(
-		'byDN'  => array(),
-		'byUid' => array(),
-	);
+	protected $usersByDN;
```
Comment (PVince81) [inline apps/user_ldap/lib/User/Manager.php]: protected ? I hope no subclass uses these

**Comment by PVince81** [comment]
Did a quick test and LDAP still works :+1:

File: apps/user_ldap/lib/User/Manager.php (hunk 1 above)
Comment (butonic) [inline apps/user_ldap/lib/User/Manager.php]: no subclasses use it.

**Comment by jvillafanez** [comment]
Makes sense 👍

**Comment by PVince81** [comment]
@butonic also backport to stable9 ?

**Comment by DeepDiver1975** [comment]
stable9: https://github.com/owncloud/core/pull/25324

**Pull request merged**